### PR TITLE
Saas 16 reconciliation fix csn

### PR DESCRIPTION
**Pull request by csnauwaert**

### CODE REVIEW

File: addons/account/static/src/js/reconciliation/reconciliation_model.js [hunk 1]
```diff
@@ -484,14 +492,19 @@ var StatementModel = BasicModel.extend({
         _.each(handles, function (handle) {
             var line = self.getLine(handle);
             var props = _.filter(line.reconciliation_proposition, function (prop) {return !prop.is_tax && !prop.invalid;});
+            if (props.length === 0) {
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_model.js]: if (props.length)

File: addons/account/static/src/js/reconciliation/reconciliation_model.js [hunk 2]
```diff
@@ -510,6 +523,28 @@ var StatementModel = BasicModel.extend({
                 return {handles: handles};
             });
     },
+    /**
+     * Return information on bank_statement_id and journal_id
+     */
+    getContext: function () {
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_model.js]: should be ordered alphabetically 

`@returns {Object} `

File: addons/account/static/src/js/reconciliation/reconciliation_model.js [hunk 3]
```diff
@@ -510,6 +523,28 @@ var StatementModel = BasicModel.extend({
...
+        var context = this.context;
+        context.journal_id = this.journal_id;
+        return this.context;
+    },
+    /**
+     * close the statement if only one statement opened and returns its id
+     */
+    closeStatement: function () {
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_model.js]: `@returns {Deferred<number>} resolves to the res_id of the closed statement`

File: addons/account/static/src/js/reconciliation/reconciliation_model.js [hunk 4]
```diff
@@ -759,6 +817,9 @@ var StatementModel = BasicModel.extend({
     _formatQuickCreate: function (line, values) {
         values = values || {};
         var account = this._formatNameGet(values.account_id);
+        var format_options = {
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_model.js]: formatOptions

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 5]
```diff
@@ -104,10 +106,12 @@ var StatementRenderer = Widget.extend(FieldManagerMixin, {
 
         if (state.valuenow === state.valuemax && !this.$('.done_message').length) {
             var dt = Date.now()-this.time;
+            console.log(state.context);
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: remove

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 6]
```diff
@@ -158,6 +162,13 @@ var StatementRenderer = Widget.extend(FieldManagerMixin, {
     },
     /**
      * @private
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: maybe a one line comment

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 7]
```diff
@@ -183,6 +194,25 @@ var StatementRenderer = Widget.extend(FieldManagerMixin, {
             view_mode: "list"
         });
     },
+    /**
+     * @private
+     * @param {MouseEvent} event
+     */
+    _onGoToBankStatement: function (e) {
+        var journal_id = $(e.target).attr('data_journal_id');
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: journalID

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 8]
```diff
@@ -183,6 +194,25 @@ var StatementRenderer = Widget.extend(FieldManagerMixin, {
             view_mode: "list"
         });
     },
+    /**
+     * @private
+     * @param {MouseEvent} event
+     */
+    _onGoToBankStatement: function (e) {
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: alphabetic order

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 9]
```diff
@@ -550,15 +591,19 @@ var LineRenderer = Widget.extend(FieldManagerMixin, {
      * @param {MouseEvent} event
      */
     _onSelectMoveLine: function (event) {
-        var mv_line_id = $(event.target).closest('.mv_line').data('line-id');
+        var $el = $(event.target)
+        this._destroyPopover($el);
+        var mv_line_id = $el.closest('.mv_line').data('line-id');
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: moveLineID

File: addons/account/static/src/js/reconciliation/reconciliation_renderer.js [hunk 10]
```diff
@@ -550,15 +591,19 @@ var LineRenderer = Widget.extend(FieldManagerMixin, {
...
     },
     /**
      * @private
      * @param {MouseEvent} event
      */
     _onSelectProposition: function (event) {
-        var mv_line_id = $(event.target).closest('.mv_line').data('line-id');
+        var $el = $(event.target)
+        this._destroyPopover($el);
+        var mv_line_id = $el.closest('.mv_line').data('line-id');
```
Comment (ged-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_renderer.js]: moveLineID

File: addons/account/static/tests/reconciliation_tests.js [hunk 11]
```diff
@@ -722,6 +735,27 @@ QUnit.module('account', {
...
+            data: this.params.data,
+        });
+
+        clientAction.appendTo($('#qunit-fixture'));
+        var widget = clientAction.widgets[0];
+        assert.strictEqual(widget.$('.o_input_dropdown input').val(), "Agrolait", "the partner many2one should display agrolait");
+        assert.strictEqual(widget.$('.match table tr').length, 2, "agrolait should have 2 propositions for reconciliation");
+
+        // Simulate changing partner
+        clientAction._onAction({target: widget, name: 'change_partner', data: {data: {display_name: 'Camptocamp', id: 12}}, stopped: false});
```
Comment (csnauwaert) [inline addons/account/static/tests/reconciliation_tests.js]: mouseenter

File: addons/account/models/account_bank_statement.py [hunk 12]
```diff
@@ -470,7 +471,7 @@ def reconciliation_widget_auto_reconcile(self, num_already_reconciled_lines):
                 'details': {
                     'name': _("Automatically reconciled items"),
                     'model': 'account.move',
-                    'ids': automatic_reconciliation_entries.mapped('journal_entry_ids').ids
+                    'ids': automatic_reconciliation_entries.mapped('journal_entry_ids').mapped('move_id').ids
```
Comment (xmo-odoo) [inline addons/account/models/account_bank_statement.py]: No need to call mapped twice, you can use a dotted path.

File: addons/account/models/account_bank_statement.py [hunk 13]
```diff
@@ -335,6 +335,7 @@ def reconciliation_widget_preprocess(self):
             'st_lines_ids': st_lines_left.ids,
             'notifications': [],
             'statement_name': len(statements) == 1 and statements[0].name or False,
+            'journal_id': statements and statements[0].journal_id.id or False,
```
Comment (xmo-odoo) [inline addons/account/models/account_bank_statement.py]: No need for the conditional, you can just write`statements[:1].journal_id.id`

File: addons/account/static/src/js/reconciliation/reconciliation_model.js [hunk 14]
```diff
@@ -558,6 +593,29 @@ var StatementModel = BasicModel.extend({
...
+    _changePartner: function (handle, partner_id) {
+        var self = this;
+        return this._rpc({
+                model: 'res.partner',
+                method: 'read',
+                args: [partner_id, ["property_account_receivable_id", "property_account_payable_id"]],
+            }).then(function (result) {
+                if (result.length > 0) {
+                    var line = self.getLine(handle);
+                    self.lines[handle].st_line.open_balance_account_id = line.amount < 0 ? result[0]['property_account_payable_id'][0] : result[0]['property_account_receivable_id'][0];
```
Comment (xmo-odoo) [inline addons/account/static/src/js/reconciliation/reconciliation_model.js]: Line breaks wouldn't hurt

**Comment by csnauwaert** [comment]
has been merged

**Pull request closed without merging**